src/webgpu/util/device_pool.ts

```diff
@@ -309,6 +312,9 @@ class DeviceHolder implements DeviceProvider {
       throw ex;
     }
 
+    // Attempt to wait for the queue to be idle.
+    await this.device.queue.onSubmittedWorkDone();
```

Comment:
This is a little silly, but since this code runs in every test, let's guard it just so implementation of onSubmittedWorkDone doesn't block all testing
```suggestion
    if (this.device.queue.onSubmittedWorkDone) {
      await this.device.queue.onSubmittedWorkDone();
    }
```